### PR TITLE
Get typeOptions from request body.

### PR DESCRIPTION
**Pull request by JSAssassin**
For other authenticationMethod like `login-email-challenge` the entryStyle is 'human'.

### CODE REVIEW

**Review by JoshDunnDev — APPROVED** [review]

File: lib/index.js [hunk 1]
```diff
@@ -59,9 +59,13 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
         // since anyone can request a nonce,
         // this is permitted without authentication
         const {actor = null} = (req.user || {});
+        let typeOptions;
+        if(authenticationMethod === 'token-client-registration') {
+          typeOptions = {entryStyle: 'machine'};
+        }
```
Comment (dlongley) [inline lib/index.js]: We should just be able to specify `typeOptions` via the client calling the API, right? Not hard code this here for a particular `authenticationMethod`?

**Review by dlongley — CHANGES_REQUESTED** [review]

File: lib/index.js (hunk 1 above)
Comment (JSAssassin) [reply lib/index.js]: Updated to get typeOptions from client.

**Review by dlongley — APPROVED** [review]
This is good, but we'll also want to validate the `typeOptions`. That doesn't have to be in this PR, but if not, we should file an issue and get it resolved soon.

**Comment by JSAssassin** [comment]
@dlongley Can the issue regarding the `typeOptions validation` be included in this already existing issue? https://github.com/digitalbazaar/bedrock-authn-token-http/issues/2

**Comment by dlongley** [comment]
> @dlongley Can the issue regarding the `typeOptions validation` be included in this already existing issue? #2

Sure!

**Review by mattcollier — APPROVED** [review]

**Pull request merged**